homeassistant/components/lyric/binary_sensor.py

```diff
@@ -0,0 +1,96 @@
+"""Support for Honeywell Lyric binary sensor platform."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+from typing import cast
+
+from aiolyric import Lyric
+from aiolyric.objects.device import LyricDevice
+from aiolyric.objects.location import LyricLocation
+from aiolyric.objects.priority import LyricRoom
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import LyricRoomEntity
+from .const import DOMAIN
+
+
+@dataclass
+class LyricRoomBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Class describing Honeywell Lyric sensor entities."""
+
+    is_on: Callable[[LyricRoom], StateType | datetime] = round
+
+
+class LyricRoomBinarySensor(LyricRoomEntity, BinarySensorEntity):
+    """Define a Honeywell Lyric binary sensor."""
+
+    coordinator: DataUpdateCoordinator[Lyric]
+    entity_description: LyricRoomBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[Lyric],
+        description: LyricRoomBinarySensorEntityDescription,
+        location: LyricLocation,
+        device: LyricDevice,
+        room: LyricRoom,
+    ) -> None:
+        """Initialize."""
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            room,
+            description.key,
+        )
+        self.entity_description = description
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return the state."""
+        room: LyricRoom = self.room
+        return cast(bool, self.entity_description.is_on(room))
```

Comment:
It would be nicer to move this around so this only returns bool (meaning please change the return type of `is_on` to bool) and place the logic at where you create the entity description